### ISSUE TITLE
Exception after defaultRouteReachability()

### ISSUE THREAD

**Issue by morisk** [ProcedureKit/ProcedureKit#360]
Hi,
I am working with Reachable operations and getting exception in  Reachability.swift.
I have my own URLSession lifted into operation and then composed into reachable operation.

* Occurs both in version 2.10.1 and new 3.0 version.
* I am using xCode 7.3.1(latest)

Here is the code where exception is raised, every time i run the code. 
```Swift
    func startNotifierOnQueue(queue: dispatch_queue_t) throws {
        assert(delegate != nil, "Reachability Delegate not set.")
        let reachability = try defaultRouteReachability()
        if !notifierIsRunning {
            notifierIsRunning = true
            var context = SCNetworkReachabilityContext(version: 0, info: nil, retain: nil, release: nil, copyDescription: nil)
            context.info = UnsafeMutablePointer(Unmanaged.passUnretained(self).toOpaque())

            guard SCNetworkReachabilitySetCallback(reachability, __device_reachability_callback, &context) else {
                stopNotifier()
                throw Error.FailedToSetNotifierCallback
            }

            guard SCNetworkReachabilitySetDispatchQueue(reachability, queue) else {
                stopNotifier()
                throw Error.FailedToSetDispatchQueue // <<------ raised here
            }
        }
        check(reachability, queue: queue)
    }
```

> will executefatal error: 'try!' expression unexpectedly raised an error: 
> Operations.Reachability.Error.FailedToSetDispatchQueue: file /Library/Caches/com.apple.xbs/Sources/swiftlang/swiftlang-703.0.18.8/src/swift/stdlib/public/core/ErrorType.swift, line 54

if I am changing to reachability `defaultRouteReachability()`  to:
```
let host = "google.com"
let reachability = reachabilityForHost(host)!
```
It works well for most of the time.

**Comment by danthorpe** [comment]
Hi @morisk - thanks for this! Will look into this tonight.

**Comment by morisk** [comment]
If I am ignoring `SCNetworkReachabilitySetDispatchQueue` false result, everything works fine.

**Comment by danthorpe** [comment]
@morisk this should now be fixed on the `swift/2.2` branch. Will be doing a 3.1.1 release for it. :)
